File: SieveOfEratosthenes/SieveOfEratosthenes.js
const math = require('mathjs')

let removeMultiples = function(i,n) {
    let array = [];
    let x = true;
    let count = 0;
    while(x==true){
        let num = math.pow(i,2) + (count*i);
        if(num<n){
            array.push(num);
            count = count + 1;
        }
        else {
            x = false;
        };
    };
    return array;
};

let sieve = function(n) {
    if(n>1) {
        let a = {};
        for(let i = 2; i<n; i++){
            a[i] = true;
        };
        for(let j = 2; j<math.round(math.sqrt(n))+1;j++) {
            if(a[j] == true) {
                let arr = removeMultiples(j,n);
                for (let k = 0; k < arr.length; k++) {
                    a[arr[k]] = false;
                };
            };
        };
        return a;
    };
};

let nums = sieve(10);
let ans = [];
console.log(Object.keys(nums));
for (let v = 2; v < Object.keys(nums).length; v++) {
    if(nums[v] == true) {
        ans.push(v);
    };
};

console.log(
    ans.reduce((a, b) => a + b, 0)
)

console.log(nums);

File: SieveOfEratosthenes/SieveOfEratosthenes.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class SieveOfEratosthenes {

    public ArrayList<Integer> removeMultiples(int i, int n) {
        ArrayList<Integer> array = new ArrayList<Integer>();
        boolean X = true;
        int count = 0;
        while (X == true) {
            int num = i * i + (count * i);
            if (num < n) {
                array.add(num);
                count = count + 1;
            } else {
                X = false;
            }
            ;
        }
        ;


        return array;
    }

    ;

    public Map<Integer, Boolean> sieve(int n) {
        ArrayList<Integer> arr = new ArrayList<Integer>();
        Map<Integer, Boolean> a = new HashMap<Integer, Boolean>();

        if (n > 1) {

            for (int x = 2; x < n; x++) {
                a.put(x, true);
            }
            ;

            for (int i = 2; i < Math.round(Math.sqrt(n)) + 1; i++) {
                if (a.get(i) == true) {
                    arr = removeMultiples(i, n);
                    for (int j : arr) {
                        a.put(j, false);
                    }
                    ;
                }
                ;
            }
            ;
            return a;
        }
        ;
        return a;
    }

    ;

    public static void main(String[] args) {

        SieveOfEratosthenes SOE = new SieveOfEratosthenes();
        Map<Integer, Boolean> ans = SOE.sieve(10);
        System.out.println(ans);
    }

    ;

};
